mmpose/models/pose_estimators/base.py

```diff
@@ -45,6 +46,12 @@ def __init__(self,
 
         self.backbone = MODELS.build(backbone)
 
+        # the PR #2108 and #2126 modified the interface of neck and head.
+        # The following function automatically detects outdated
+        # configurations and updates them accordingly, while also providing
+        # clear and concise information on the changes made.
+        neck, head = check_and_update_config(neck, head)
```

Comment:
I think this function should also take the backbone cfg as input, so that it can be used as a more general check and auto-update process for the entire model, rather than just for compatibility measures with the previous PR.

Reply:
When trying to inference a model with `SyncBN` on `cpu`, it will raise an error. I think we can check the device and automatically repalce `SyncBN` as `BN` in the config dict.

Reply:
> When trying to inference a model with `SyncBN` on `cpu`, it will raise an error. I think we can check the device and automatically repalce `SyncBN` as `BN` in the config dict.

`SyncBN` can be converted to `BN` by `mmengine.model.revert_sync_batchnorm`. The Inferencer has supported this conversion in @2152

Reply:
> I also have a concern that at the beginning of the training, `mmengine` will print the entire config and dump a config file. Will this automatically processed function affect the above content? Such automatic modification seems to cause differences between the real model structure and the output. Do you think there is any way to optimize it?

We can modify this checker to a hook in the future, and add more functions to it.